### PR TITLE
User story 43

### PR DESCRIPTION
**Pull request by ggerdsen**
Added button to activate the inactive items and associated tests.

### CODE REVIEW

**Comment by NickEdwin** [comment]
Thanks for fixing my broken tests! :)

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -7,10 +7,14 @@ def index
 
   def update
     item = Item.find(params[:id])
-    item.toggle(:active?)
-    if item.save
+    if item.active?
```
Comment (sienna-kopf) [inline app/controllers/merchant/items_controller.rb:10]: active? vs enabled? is confusing!

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -7,10 +7,14 @@ def index
...
+      item.update(active?: false)
       flash[:success] = "#{item.name} is no longer for sale"
       redirect_to "/merchant/items"
+    else
+      item.update(active?: true)
```
Comment (sienna-kopf) [inline app/controllers/merchant/items_controller.rb:15]: I wonder why toggle doesn't work now....

File: app/views/merchant/items/index.html.erb [hunk 3]
```diff
@@ -14,9 +14,10 @@
         <p>Inventory: <%= item.inventory %> </p>
         <% if item.active? %>
           <p>Active</p>
-          <%= button_to "Deactivate", "/merchant/#{item.id}", method: :patch %>
+          <%= button_to "Deactivate", "/merchant/#{item.id}/update", method: :patch %>
```
Comment (sienna-kopf) [inline app/views/merchant/items/index.html.erb:17]: specifying this route as update makes the route explicit!

File: config/routes.rb [hunk 4]
```diff
@@ -57,7 +57,7 @@
     get "/", to: "dashboard#index"
     get "/orders/:order_id", to: "orders#show"
     get "/items", to: "items#index"
-    patch "/:id", to: "items#update"
+    patch "/:id/update", to: "items#update"
```
Comment (sienna-kopf) [inline config/routes.rb:60]: Nice & Specific!

File: spec/features/merchants/items_spec.rb [hunk 5]
```diff
@@ -27,13 +27,36 @@
...
+      end
+      expect(current_path).to eq("/merchant/items")
+
+      within "#item-#{@tire.id}" do
+        click_on "Activate"
+      end
+      
+      expect(current_path).to eq("/merchant/items")
+      
+      @meg.items.all? {|item| item.active? == true}
```
Comment (sienna-kopf) [inline spec/features/merchants/items_spec.rb:56]: Changing this here to item.active? makes the test far more clear!

**Pull request merged**